src/client/datascience/jupyter/jupyterSession.ts

```diff
@@ -275,7 +277,11 @@ export class JupyterSession extends BaseJupyterSession {
             } catch (ex) {
                 // If we failed to create the kernel, we need to clean up the file.
                 if (this.connInfo && backingFile) {
-                    this.contentsManager.delete(backingFile.path).ignoreErrors();
+                    void this.contentsManager
+                        .delete(backingFile.path)
+                        .catch((ex) => traceWarning(`Failed to delete ${backingFile.path} using Jupyter API`, ex))
+                        // Sometimes the files don't get deleted via Jupyter, delete them manually.
+                        .finally(() => this.fs.deleteLocalFile(backingFile.path));
```

Comment:
Why don't we do this always instead of using the Jupyter lab API
I'm not sure why we need to use the Jupyter API, perhaps it has some better cleanup?
This way seems safe

Reply:
Jupyter works in the remote scenario.

I believe the checkpoint option needs to be turned off too, otherwise the temporary file will keep coming back.

Reply:
this code aoplies to local, however checkpoints make sense